### PR TITLE
migration perfscale job to prow ci for nutanix

### PR DESCRIPTION
**Pull request by liqcui**
/hold

### CODE REVIEW

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  steps:
+    post:
+    - chain: cucushift-installer-rehearse-nutanix-ipi-deprovision    
+    pre:
+    - chain: ipi-conf-nutanix
+    - ref: ipi-conf-osimage-nutanix
+    - ref: ipi-conf-nutanix-customized-resource
+    - chain: ipi-install
+    - ref: configure-registry-storage-nutanix
+    - chain: cucushift-installer-check
```
Comment (jianlinliu) [inline ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:12]: You are trying to launch a simple IPI nutanix cluster? If yes, pls call `cucushift-installer-rehearse-nutanix-ipi-provision` chain directly.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- always_run: false
+  as: control-plane-24nodes
+  steps:
+    allow_skip_on_success: true
+    cluster_profile: nutanix-qe
+    test:
+    - chain: openshift-qe-control-plane
+    workflow: cucushift-installer-rehearse-nutanix-ipi
+- always_run: false
+  as: data-path-9nodes
```
Comment (jianlinliu) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:34]: As far as I know, nutanix cluster is only can be installed on `cluster: build01` build farm. cc @sgaoshang to confirm.

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:12]: `cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource-provision` support to setting replica via `COMPUTE_REPLICAS` ENV, so you can reference the chain.

Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:12]: Thanks Jianlin, I will remove useless steps once I provision a nutanix clusters with 24 worker nodes. I will try your suggestion first.

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-node-density-heavy-24nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-control-plane-24nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml (hunk 2 above)
Comment (liqcui) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:38]: This is the key, we can create nutanix cluster after adding cluster: build01, thanks!

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml (hunk 1 above)
Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:12]: It report the error  ` /bin/bash: line 35: prism_element1_uuid: unbound variable` when I use cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource-provision, I use partial of chain in cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource-provision in my customized workflow

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    test:
+    - chain: openshift-qe-control-plane
+    workflow: openshift-qe-installer-nutanix
+- always_run: false
+  as: data-path-9nodes
+  cluster: build01
+  steps:
+    allow_skip_on_success: true
+    cluster_profile: nutanix-qe
+    env:
```
Comment (qiliRedHat) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:43]: @liqcui For data path test, we use 8 vCPUs 32 GiB size of worker nodes. But I don't see where we can configure the worker node size in ref ipi-conf-nutanix. Could you help to double confirm with installer team?

**Review by qiliRedHat — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+workflow:
+  as: openshift-qe-installer-nutanix
+  steps:
+    post:
+    - chain: cucushift-installer-rehearse-nutanix-ipi-deprovision    
+    pre:
+    - chain: ipi-conf-nutanix
+    - ref: ipi-conf-nutanix-customized-resource
+    - chain: ipi-install
+    - ref: configure-registry-storage-nutanix
```
[the remark this answers is not in the document]
Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:10]: It report the error  /bin/bash: line 35: prism_element1_uuid: unbound variable when I use cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource-provision, I use partial of chain in cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource-provision in my customized workflow
@jianlinliu Do you know how to fix the error?

[the remark this answers is not in the document]
Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml:10]: @qiliRedHat you mean remove **- ref: configure-registry-storage-nutanix**, all all steps?

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml (hunk 3 above)
Comment (liqcui) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:43]: It's in ipi-conf-nutanix-customized-resource-ref.yaml, updated worker node to 8 vCPUs 32 GiB

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    - ref: enable-qe-catalogsource
```
[the remark this answers is not in the document]
Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml]: removed now

File: ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (liqcui) [reply ci-operator/step-registry/openshift-qe/installer/nutanix/openshift-qe-installer-nutanix-workflow.yaml]: Removed now

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml (hunk 3 above)
Comment (qiliRedHat) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:43]: @liqcui Thanks for the info. Then let's also specify the infra node size with SET_ENV_BY_PLATFORM=custom and the below env vars. For infra we used 16 vCPUs 64 GiB
OPENSHIFT_INFRA_NODE_INSTANCE_VCPU=16
OPENSHIFT_INFRA_NODE_INSTANCE_MEMORYSIZE=64Gi

Comment (paigerube14) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:43]: do we need that worker cpu and memory size as well as the infra cpu and memory size for all the tests listed here?

**Comment by paigerube14** [comment]
After adding the infra cpu/memory specifications are you able to also copy the config for 4.17 so we can use this for our 4.17 regression testing

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

**Comment by liqcui** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-nutanix-4.16-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml (hunk 3 above)
Comment (liqcui) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.16-nightly-x86.yaml:43]: @paigerube14 @qiliRedHat I have adding new job for 4.17 also, this time I only add above three ENV in datapath job

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.17-nightly-x86.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+build_root:
+  image_stream_tag:
+    name: ci-tools-build-root
+    namespace: ci
+    tag: latest
+images:
+- dockerfile_path: prow/Dockerfile
+  to: ocp-qe-perfscale-ci-nutanix-417
+promotion:
```
Comment (paigerube14) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.17-nightly-x86.yaml:9]: we don't need the images or promotion section here, its just making an un-needed postsubmit job

Comment (liqcui) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__nutanix-4.17-nightly-x86.yaml:6]: Removed now,thanks!

**Pull request merged**